plugins/modules/opensearch.py

```diff
@@ -948,6 +948,7 @@ def set_advanced_security_options(
             ] = advanced_security_opts.get("internal_user_database_enabled")
         master_user_opts = advanced_security_opts.get("master_user_options")
         if master_user_opts is not None:
+            advanced_security_config["MasterUserOptions"] = {}
```

Comment:
```suggestion
            if not advanced_security_config.get("MasterUserOptions"):
                advanced_security_config["MasterUserOptions"] = {}
```

I'm not sure about it....but I feel better if the empty dict is only set, if `MasterUserOptions` key doesn't exist.  
cc @alinabuzachis